tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/Diagnostics/InvalidVersionFormatDiagnostic.cs

```diff
@@ -0,0 +1,27 @@
+﻿// <copyright file="InvalidVersionFormatDiagnostic.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Microsoft.CodeAnalysis;
+
+namespace Datadog.Trace.SourceGenerators.InstrumentationDefinitions.Diagnostics;
+
+internal static class InvalidVersionFormatDiagnostic
+{
+    // internal for testing
+    internal const string Id = "ID2";
+    private const string Title = "Invalid version format";
+    private const string Message = "Version must be of form <major>.<minor>.<patch> with optional * for 'any'";
+
+    public static Diagnostic Create(SyntaxNode? currentNode) =>
+        Diagnostic.Create(
+            new DiagnosticDescriptor(
+                Id,
+                Title,
+                Message,
+                category: TagsListGenerator.Constants.Usage,
```

Comment:
nit: The Usage constant could be refactored to `Datadog.Trace.SourceGenerators.Constants.Usage`

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2288/commits/bb06fe0e6040cc2707f475d33e03a3e300719ddf